DOCUMENTATION.md

```diff
@@ -43,6 +45,153 @@ The overall territory is laid out here in this [C4](https://c4model.com) System
 
 ![Propulsion c4model.com Container Diagram: Reactors](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.github.com/jet/propulsion/master/diagrams/ReactorsContainer.puml&fmt=svg)
 
+# The `Propulsion.Streams` Programming Model for Projections, Reactions and Workflows
+
+`Propulsion.Streams` provides a programming model to manage running of _Handlers_ in a manner that optimises for the following:
+- isolating the handlers from the client libraries of a given Event Store (the Handler is triggered via a 'Sink' that manages accepts incoming events and checkpointing of progress in a Store-specific manner relevant to the hosting environment in which your handler will be run).
+- providing a clean approach to the testing of Reaction logic with and without involving your actual Event Store (the `MemoryStoreProjector` component is a key part of that story).
+- exposing thorough diagnostic information that enables one to effectively monitor and tune your Handlers in the context of an overall production system.
+
+## Overview of running projections with Propulsion
+
+### _Source pipeline_
+
+'Source' refers to a set of components that comprise the store-specific 'consumer' side of the processing. It encompasses:
+   a. `Source`: the top level component
+      - maintains a reference to the `Sink`, into which the `Ingester`s will feed batches of events
+      - controls the set of Tranches from which events will be consumed (for a store like EventStore, there's only a single Tranche representing the `'$all'` stream; for CosmosStore, there's a Tranche per physical partition in the Container etc)
```

Comment:
To consider: add a terminology section that explains a tranche is a portion/division of a pool (literally means slice in french). In this case the pool is the pool of all events.

I remember being confused by tranche, having not seen it before 😄

Reply:
Good point, thanks (added Glossary and inputs commit)